### DISCUSSION TITLE
xinput_calibrator equivalent

### DISCUSSION THREAD

**Discussion by matlawicebox** [WayfireWM/wayfire#2050] [Q&A]
I can't find any information on how to apply scaling calibration for a touchscreen under wayfire.  

I'm using bookwork on a raspberry pi and under the default wayfire, I couldn't get my touchscreen to respond correctly - response to touch was offset in x and y (but not rotated or inverted).  I've looked for information on how to do this and found information on inverting and combining screens together, but nothing on applying a scaling factor.

In the end I got my touchscreen to work by switching to X11 and then using xinput_calibrator, but if the raspberry pi OS eventually discontinues X11 support this is only a temporary fix.

Ideally there would be inputs in ~/.config/wayfire.ini similar to the xmax/xmin/ymax/ymin inputs in /etc/X11/xorg.conf.d/99-calibration.conf.

Do those already exist?  If so, could someone point me in the direction of how to determine the correct input values and how to add them appropriately to the ini file?

**Comment by ammen99** [comment]
Take a look at my answer here: https://github.com/WayfireWM/wayfire/issues/1999
